Cross-Compiling Tao For MacOS On Linux

I have an initial discussion with @joseluisq [here](https://github.com/joseluisq/rust-linux-darwin-builder/issues/9)

I am trying to compile a [project](https://github.com/themeliolabs/ginkou-loader) that depends on tao.

The issue comes down to building the [carbon_hotkey files](https://github.com/tauri-apps/tao/tree/dev/src/platform_impl/macos/carbon_hotkey).

On a `macos-latest` machine, I ran these two steps:
```
        - name: Clone Tao
          run: git clone https://github.com/tauri-apps/tao

        - name: Build
          run: pushd tao/src/platform_impl/macos/carbon_hotkey; clang -v -mmacosx-version-min=10.12 carbon_hotkey_binding.c; popd
```

This outputs the same error when trying to compile on linux. The message from the runner job is this:
```
Apple clang version 13.0.0 (clang-1300.0.29.30)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
 "/Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang" -cc1 -triple x86_64-apple-macosx10.12.0 -Wundef-prefix=TARGET_OS_ -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj -mrelax-all --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name carbon_hotkey_binding.c -mrelocation-model pic -pic-level 2 -mframe-pointer=all -fno-strict-return -fno-rounding-math -munwind-tables -faligned-alloc-unavailable -target-sdk-version=12.1 -fvisibility-inlines-hidden-static-local-var -target-cpu penryn -tune-cpu generic -debugger-tuning=lldb -target-linker-version 711 -v -resource-dir /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0 -isysroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -I/usr/local/include -internal-isystem /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include -internal-isystem /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/include -internal-externc-isystem /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include -internal-externc-isystem /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include -Wno-reorder-init-list -Wno-implicit-int-float-conversion -Wno-c99-designator -Wno-final-dtor-non-final-class -Wno-extra-semi-stmt -Wno-misleading-indentation -Wno-quoted-include-in-framework-header -Wno-implicit-fallthrough -Wno-enum-enum-conversion -Wno-enum-float-conversion -Wno-elaborated-enum-base -fdebug-compilation-dir /Users/runner/work/mellis/mellis/tao/src/platform_impl/macos/carbon_hotkey -ferror-limit 19 -stack-protector 1 -mdarwin-stkchk-strong-link -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcommon -clang-vendor-feature=+nullptrToBoolConversion -clang-vendor-feature=+messageToSelfInClassMethodIdReturnType -clang-vendor-feature=+disableInferNewAvailabilityFromInit -clang-vendor-feature=+disableNeonImmediateRangeCheck -clang-vendor-feature=+disableNonDependentMemberExprInCurrentInstantiation -fno-odr-hash-protocols -clang-vendor-feature=+revert09abecef7bbf -mllvm -disable-aligned-alloc-awareness=1 -mllvm -enable-dse-memoryssa=0 -o /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/carbon_hotkey_binding-8edd91.o -x c carbon_hotkey_binding.c
clang -cc1 version 13.0.0 (clang-1300.0.29.30) default target x86_64-apple-darwin20.6.0
ignoring nonexistent directory "/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/local/include"
ignoring nonexistent directory "/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/include
 /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include
 /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/System/Library/Frameworks (framework directory)
End of search list.
 "/Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld" -demangle -lto_library /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/libLTO.dylib -no_deduplicate -dynamic -arch x86_64 -platform_version macos 10.12.0 12.1 -syslibroot /Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk -o a.out -L/usr/local/lib /var/folders/24/8k48jl6d249_n_qfxwsl6xvm0000gn/T/carbon_hotkey_binding-8edd91.o -lSystem /Applications/Xcode_13.2.1.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/clang/13.0.0/lib/darwin/libclang_rt.osx.a
Undefined symbols for architecture x86_64:
  "_GetApplicationEventTarget", referenced from:
      _install_event_handler in carbon_hotkey_binding-8edd91.o
      _register_hotkey in carbon_hotkey_binding-8edd91.o
  "_GetEventParameter", referenced from:
      _hotkey_handler in carbon_hotkey_binding-8edd91.o
  "_InstallEventHandler", referenced from:
      _install_event_handler in carbon_hotkey_binding-8edd91.o
  "_RegisterEventHotKey", referenced from:
      _register_hotkey in carbon_hotkey_binding-8edd91.o
  "_RemoveEventHandler", referenced from:
      _uninstall_event_handler in carbon_hotkey_binding-8edd91.o
  "_UnregisterEventHotKey", referenced from:
      _unregister_hotkey in carbon_hotkey_binding-8edd91.o
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Error: Process completed with exit code 1.
```

Is there something simple I am missing here?

--- a/.changes/macos-crosscompile.md
+++ b/.changes/macos-crosscompile.md
@@ -1,0 +1,5 @@
+---
+"wry": patch
+---
+
+Support cross compiling for macos from a non macos host.

--- a/.changes/macos-crosscompile.md
+++ b/.changes/macos-crosscompile.md
@@ -1,5 +1,5 @@
 ---
-"wry": patch
+"tao": patch
 ---
 
 Support cross compiling for macos from a non macos host.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -31,6 +31,9 @@ default = [ ]
 dox = [ "gtk/dox" ]
 tray = [ "libappindicator", "dirs-next" ]
 
+[build-dependencies]
+cc = "1"
+
 [dependencies]
 instant = "0.1"
 lazy_static = "1"
@@ -63,9 +66,6 @@ core-graphics = "0.22"
 dispatch = "0.2"
 scopeguard = "1.1"
 png = "0.17"
-
-[target."cfg(target_os = \"macos\")".build-dependencies]
-cc = "1"
 
 [target."cfg(target_os = \"windows\")".dependencies]
 parking_lot = "0.12"

--- a/build.rs
+++ b/build.rs
@@ -12,7 +12,6 @@ fn main() {
     println!("cargo:rustc-cfg=use_colorsync_cgdisplaycreateuuidfromdisplayid");
   }
   // link carbon hotkey on macOS
-  #[cfg(target_os = "macos")]
   {
     if std::env::var("CARGO_CFG_TARGET_OS").map_or(false, |os| os == "macos") {
       println!("cargo:rustc-link-lib=framework=Carbon");

